langchain/src/chains/base.ts

```diff
@@ -98,8 +97,7 @@ export abstract class BaseChain implements ChainInputs {
    * Wraps {@link _call} and handles memory.
    */
   async call(values: ChainValues): Promise<ChainValues> {
-    const fullValues = deepcopy(values);
-
+    const fullValues = values;
```

Comment:
@agola11 @nfcampos this was giving me issues. basically deepcopy was not respecting the types of the objects, and was convertiing HumanChatMessage just to generic `{"text": "foo"}` object, which was leading to issues down the road

i **think** this deepcopy is not super important... but if theres a better way to do this, just to be safe, it might be good...

Reply:
I don't think the deep copy is needed either, I think the fix would be `const fullValues = deepcopy(values) as typeof values;` but I couldn't reproduce whatever issue you were facing @hwchase17